bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/update/RemoveChannelInstructionImpl.java

```diff
@@ -40,7 +47,11 @@ public int getThingTypeVersion() {
 
     @Override
     public void perform(Thing thing, ThingBuilder thingBuilder) {
-        ChannelUID affectedChannelUid = new ChannelUID(thing.getUID(), channelId);
-        thingBuilder.withoutChannel(affectedChannelUid);
+        if (groupIds.isEmpty()) {
+            thingBuilder.withoutChannel(new ChannelUID(thing.getUID(), channelId));
+        } else {
+            groupIds.forEach(
+                    groupId -> thingBuilder.withoutChannel(new ChannelUID(thing.getUID(), groupId + "#" + channelId)));
```

Comment:
I don't know how much this is actually being used, so just for consideration:
```suggestion
                    groupId -> thingBuilder.withoutChannel(new ChannelUID(thing.getUID(), groupId + ChannelUID.CHANNEL_GROUP_SEPARATOR + channelId)));
```